arch/arm/src/sama5/sam_tsd.c

```diff
@@ -1655,7 +1735,10 @@ int sam_tsd_register(struct sam_adc_s *adc, int minor)
 
   /* Initialize the touchscreen device driver instance */
 
-  priv->adc = adc; /* Save the ADC device handle */
+  priv->adc     = adc;               /* Save the ADC device handle    */
+  priv->threshx = INVALID_THRESHOLD; /* Initialize thresholding logic */
+  priv->threshy = INVALID_THRESHOLD; /* Initialize thresholding logic */
+  nxsem_init(&priv->waitsem, 0, 0);
```

Comment:
Why those fields can't be initialized statically?

Reply:
Wasn't my code or change. But can look at this.